From 21f04792fa9785a1dbed01ff4702e07d8111ea1c Mon Sep 17 00:00:00 2001
From: Chinpo Nya <czen+github@honk.li>
Date: Mon, 16 Oct 2023 03:46:12 +0200
Subject: [PATCH 1/4] http client fix

---
 S/LIB/SOCKET.LUA | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/S/LIB/SOCKET.LUA b/S/LIB/SOCKET.LUA
index 007ef7d..a742ba2 100644
--- a/S/LIB/SOCKET.LUA
+++ b/S/LIB/SOCKET.LUA
@@ -73,14 +73,14 @@ TCPBASE = {
 			table.insert(send, "");
 			table.insert(send, param);
 		else
-			table.insert(send, "\n");
+			table.insert(send, "\r\n");
 		end
 
 		self.uuid = getUUID();
 		
 _dm( string.format("query[%s]<br>%s<br>\n", self.uuid, table.concat(send, "<br>\n")));
 
-		self.stream = table.concat(send, "\n");
+		self.stream = table.concat(send, "\r\n");
 		self.socket:setStream( self.stream );
 
 		return 0;
-- 
2.44.1

